conwayste/src/ui/textfield.rs

```diff
@@ -243,10 +243,10 @@ impl Widget for TextField {
             let text_with_cursor = self.text.clone();
             let text_pos = Point2::new(self.dimensions.x + CURSOR_OFFSET_PX, self.dimensions.y);
 
-            draw_text(ctx, font, Color::from(css::WHITESMOKE), &text_with_cursor, &text_pos, None)?;
+            draw_text(ctx, font, Color::from(css::WHITESMOKE), &text_with_cursor[self.visible_start_index..self.visible_end_index], &text_pos, None)?;
```

Comment:
It might be a good idea to have variables or constants for the colors of text when highlighted, and normal, so that the actual colors are not mixed everywhere in the code. Just a thought :grin: